gorm-tools-csv/src/main/groovy/gorm/tools/csv/CSVPathKeyMapReader.groovy

```diff
@@ -56,7 +56,7 @@ class CSVPathKeyMapReader extends CSVReaderHeaderAware {
         Map data = super.readMap()
         data = prune && data ? Maps.prune(data) : data
         // data = data as Map<String, String>
-        return PathKeyMap.of(data, pathDelimiter) as Map<String, Object>
+        return PathKeyMap.of(data, pathDelimiter).init() as Map<String, Object>
```

Comment:
We want to delay the init until its used for performance reasons we found in the benchmarks. This will eliminate the lazy loading and defeat the purpose of it. readAllRows ends up calling this which means its called for every row.